src/channel_state.ts

```diff
@@ -86,19 +83,19 @@ export class ChannelState<
           MessageType,
           ReactionType,
           UserType
-        >['messageToImmutable']
+        >['formatMessage']
       >
     >;
-  }>;
-  mutedUsers: Immutable.ImmutableArray<UserResponse<UserType>>;
-  watchers: Immutable.ImmutableObject<{
-    [key: string]: Immutable.Immutable<UserResponse<UserType>>;
-  }>;
-  members: Immutable.ImmutableObject<{
-    [key: string]: Immutable.Immutable<ChannelMemberResponse<UserType>>;
-  }>;
+  };
+  mutedUsers: Array<UserResponse<UserType>>;
+  watchers: {
+    [key: string]: UserResponse<UserType>;
+  };
+  members: {
+    [key: string]: ChannelMemberResponse<UserType>;
+  };
```

Comment:
Record is a bit cleaner
```suggestion
  watchers: Record<string, UserResponse<UserType>>;
  members: Record<string, ChannelMemberResponse<UserType>>;
```